### PR TITLE
Make timeline flame chart section labels sticky

### PR DESCRIPTION
**Pull request by kenzieschmoll**
transparent_stickly_labels
transparent_sticky_labels_dark

- Make labels sticky to the right
- paint alternating colors for section backgrounds (same alt colors we use for tables)
- addresses part of https://github.com/flutter/devtools/issues/1964

### CODE REVIEW

**Comment by jacob314** [comment]
Having the labels sticky is great. Discussed offline tweaking them to use horizontal real-estate a more efficiently.

**Comment by kenzieschmoll** [comment]
Updated and ready for review. PTAL @jacob314

**Comment by jacob314** [comment]
High level thoughts:
1. Maybe increase the alpha slightly so that the labels are a little more readable.
2. For blocks with more than one line consider placing the label completely on its own line. That matches what chrome devtools does.
3. maybe add padding so that time zero doesn't overlap the labels?

File: packages/devtools_app/lib/src/common_widgets.dart [hunk 1]
```diff
@@ -767,3 +767,27 @@ extension ThemeDataExtension on ThemeData {
...
+  final color = theme.canvasColor;
+  return _modifyColorForIndex(color, index, isDarkTheme: theme.isDarkTheme);
+}
+
+Color alternatingColorForIndex(int index, {@required bool isDarkTheme}) {
+  final color = defaultBackgroundColor;
+  return _modifyColorForIndex(color, index, isDarkTheme: isDarkTheme);
+}
+
+Color _modifyColorForIndex(
```
Comment (jacob314) [inline packages/devtools_app/lib/src/common_widgets.dart:783]: nit: remove word `modify`?

File: packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart [hunk 2]
```diff
@@ -34,15 +36,17 @@ class TimelineFlameChart extends FlameChart<TimelineData, TimelineEvent> {
...
-    // TODO(kenz): we need to calculate start inset based on the width of the
-    // section labels. We should also set a max, ellipsize, and rely on tooltip
-    // to give the full name in the event that the section name exceeds max.
-    //
-    // Alternatively, we could make the label section a column of it's own that
-    // is resizeable. It would need to link scroll controllers with the list
-    // view holding the flame chart nodes. This would make section labels sticky
-    // to the left as an inherent bonus.
-    return 140.0;
+    const maxStartInset = 300.0;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart:39]: nice. I was suggesting this but looks like you already did it. :)

File: packages/devtools_app/lib/src/common_widgets.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/common_widgets.dart]: done

**Comment by kenzieschmoll** [comment]
added TODOs for some of your suggestions as some are more involved changes that should be done in a separate PR

**Pull request merged**